src/modules/core/containers/bottom-fixed-content/bottom-fixed-content.tsx

```diff
@@ -0,0 +1,119 @@
+'use client';
+import {
+  type FC,
+  type ReactNode,
+  createContext,
+  useEffect,
+  useRef,
+  useState,
+  useCallback,
+  useContext,
+  useId,
+} from 'react';
+// types
+import type { ChildrenProp } from '@/modules/core/types/react.types';
+
+import type { BottomFixedContentProps } from './bottom-fixed-content.interface';
+import styles from './bottom-fixed-content.module.scss';
+
+type RenderElement = {
+  id: string;
+  element: ReactNode;
+  orderIndex: number;
+};
+
+const BottomFixedContentContext = createContext<{
+  render: (data: RenderElement) => void;
+  unmount: (id: string) => void;
+}>({
+  render: () => {},
+  unmount: () => {},
+});
+
+const BottomFixedContent: FC<BottomFixedContentProps> & {
+  Item: FC<ChildrenProp & { orderIndex: number }>;
+} = ({ children }) => {
+  const [items, setItems] = useState<RenderElement[]>([]);
+  // refs
+  const rootRef = useRef<HTMLDivElement>(null);
+
+  useEffect(() => {
+    const element = rootRef.current;
+
+    if (!element) return;
+
+    const heightObserver = new ResizeObserver((entries) => {
```

Comment:
Isn't there any other way to do it? if not can you add comment to refactor it later, I feel like there could be another way.

Reply:
This observer is added to update global css variable `--bottom-fixed-content-height`. This variables is useful for layouts or pages where we need to add bottom padding the same as bottom fixed content height.